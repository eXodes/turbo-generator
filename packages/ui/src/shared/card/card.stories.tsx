import type { Meta, StoryObj } from "@storybook/react";

import { Card } from "./card";

const meta = {
  title: "Shared/Card",
  component: Card,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Card>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    title: "Card",
    href: "#",
    children: "Card Content",
  },
};
