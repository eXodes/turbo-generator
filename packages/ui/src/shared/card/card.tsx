"use client";

import type { JSX, ReactNode } from "react";

interface CardProps {
  title: string;
  href: string;
  children: ReactNode;
  className?: string;
}

const Card = ({ title, href, children, className }: CardProps): JSX.Element => {
  return (
    <a
      className={className}
      href={`${href}?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo"`}
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        {title} <span>-&gt;</span>
      </h2>
      <p>{children}</p>
    </a>
  );
};

export { type CardProps, Card };
