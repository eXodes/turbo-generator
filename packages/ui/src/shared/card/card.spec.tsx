import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";

import { Card } from "./card";

describe("Card", () => {
  it("should have 'Card' page title", async () => {
    const { baseElement } = render(
      <Card title={"Card Title"} href={"#"}>
        Content
      </Card>,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
