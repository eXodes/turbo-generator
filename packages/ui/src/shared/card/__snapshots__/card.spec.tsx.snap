// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > should have 'Card' page title 1`] = `
<body>
  <div>
    <a
      href="#?utm_source=create-turbo&utm_medium=basic&utm_campaign=create-turbo""
      rel="noopener noreferrer"
      target="_blank"
    >
      <h2>
        Card Title
         
        <span>
          -&gt;
        </span>
      </h2>
      <p>
        Content
      </p>
    </a>
  </div>
</body>
`;
